ethcore/spec/src/spec.rs

```diff
@@ -264,15 +265,29 @@ impl From<ethjson::spec::HardcodedSync> for SpecHardcodedSync {
 	}
 }
 
+impl Serialize for SpecHardcodedSync {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+        {
+            let mut serialized = serializer.serialize_struct("SpecHardcodedSync", 3)?;
+
+            let header_str = format!("{:x}", self.header);
+            serialized.serialize_field("header", &header_str)?;
+
+            let total_difficulty_str = format!("{:?}", self.total_difficulty);
+            serialized.serialize_field("totalDifficulty", &total_difficulty_str)?;
+
+            serialized.serialize_field("CHTs", &self.chts)?;
+            serialized.end()
+        }
+}
```

Comment:
I would use a derived implementation for serde with custom attributes like 
https://serde.rs/field-attrs.html#rename
https://serde.rs/field-attrs.html#serialize_with
as it's more future-proof e.g. when adding new fields.
But it'll probably be also more code, so it's fine as is.

Reply:
Ah good idea -- it didn't turn out to even be too much more code. Added in latest update